js/src/utils/external.ts

```diff
@@ -0,0 +1,77 @@
+
+import { spawn } from "child_process";
+import { IS_DEVELOPMENT_OR_CI, TELEMETRY_URL } from "../sdk/utils/constants";
+import { ifObjectStringify } from "../sdk/utils/common";
+
+/**
+ * Sends a reporting payload to the telemetry server using a child process.
+ * This function is intended for use in Node.js environments.
+ * 
+ * @param {any} reportingPayload - The payload to be sent to the telemetry server.
+ */
+export async function sendProcessReq(info:{
+    url: string,
+    method: string,
+    headers: Record<string, string>,
+    data: Record<string, unknown>
+}) {
+     if(IS_DEVELOPMENT_OR_CI){
+        console.log(`Hitting ${info.url}[${info.method}] with ${ifObjectStringify(info.data)}`);
+        return true;
+    }
+    try {
+        // Spawn a child process to execute a Node.js script
+        const child = spawn('node', ['-e', `
+        const axios = require('axios');
```

Comment:
The sendProcessReq function contains multiple critical security vulnerabilities. It constructs a command for child_process.spawn using unescaped user-supplied inputs (info.url, info.data, and info.headers), which can lead to command injection attacks if these values contain malicious content or special characters. This allows for the possibility of executing arbitrary code, posing a severe security risk.